### ISSUE TITLE
Add NcML backend for header_extract extraction method plugin

### ISSUE THREAD

**Issue by huard** [cedadev/asset-scanner#39]
I've written some code to parse the XML generated by THREDDS' NCML service or `ncdump -x <filepath>` that I think could be converted to a backend for the `header_extract` plugin. 

My proposal is to create `/asset_scanner/plugins/extraction_methods/header_extract/backends/ncml.py` that would parse the XML using `lxml.xpath` to find the value of attributes. 

One potential advantage of this approach is that the XML files can be cached, potentially speeding up considerably the extraction of metadata attributes. 

Before preparing a PR, I just wanted to make sure this fits with your development roadmap. 

Regards,

**Comment by agstephens** [comment]
Hi @huard, thanks for raising this issue. I think your suggested approach makes sense. However, I'm asking @rhysrevans3 and @Mahir-Sparkess to give their input as they are closer to the code than I am.

**Comment by rhysrevans3** [comment]
@huard, yes that sounds like it fits with our current development as we don't have any plans to change how the header_extract will work.

**Comment by huard** [comment]
Thanks, I have a first draft, but it's not clear how I should test it. Is is possible that `header_extract` is only tested by `item-generator` using hardcoded `badc` links?
